packages/wrangler/src/__tests__/generate.test.ts

```diff
@@ -62,30 +53,183 @@ describe("generate", () => {
+			await expect(
+				runWrangler("generate my-worker caass/wrangler-generate-test-template")
+			).resolves.toBeUndefined();
+
+			expect(readDirectory("my-worker")).toMatchObject<Directory>({
+				".git": expect.any(Object),
+				".gitignore": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+		});
+
+		it("clones a user/repo/path/to/subdirectory template", async () => {
+			await expect(
+				runWrangler("generate my-worker cloudflare/templates/worker-typescript")
+			).resolves.toBeUndefined();
+
+			expect(readDirectory("my-worker")).toMatchObject<Directory>({
+				".git": expect.any(Object),
+				".gitignore": expect.any(String),
+				"README.md": expect.stringContaining("Template: worker-typescript"),
+				"jest.config.json": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				test: expect.objectContaining({
+					"index.test.ts": expect.any(String),
+				}),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+		});
+
+		it.skip("clones a git@github.com/user/repo template", async () => {
+			await expect(
+				runWrangler(
+					"generate my-worker git@github.com:caass/wrangler-generate-test-template"
+				)
+			).resolves.toBeUndefined();
+
+			expect(readDirectory("my-worker")).toMatchObject<Directory>({
+				".git": expect.any(Object),
+				".gitignore": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+		});
+
+		it.skip("clones a git@github.com/user/repo/path/to/subdirectory template", async () => {
+			await expect(
+				runWrangler(
+					"generate my-worker git@github.com:cloudflare/templates/worker-typescript"
+				)
+			).resolves.toBeUndefined();
+
+			expect(readDirectory("my-worker")).toMatchObject<Directory>({
+				".git": expect.any(Object),
+				".gitignore": expect.any(String),
+				"README.md": expect.stringContaining("Template: worker-typescript"),
+				"jest.config.json": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				test: expect.objectContaining({
+					"index.test.ts": expect.any(String),
+				}),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+		});
 	});
 });
+
+type FileName = string;
+type FileContents = string;
+type Directory = { [key: FileName]: FileContents | Directory };
+
+function readDirectory(directoryPath: string): Directory {
```

Comment:
`init.test.ts` has a helper called `checkFiles()`, which I think does basically the same thing as this. Hope about reusing that somehow?